client/boot/app.js

```diff
@@ -0,0 +1,59 @@
+// Initialize localStorage polyfill before any dependencies are loaded
+require( 'lib/local-storage' )();
+if ( process.env.NODE_ENV === 'development' ) {
+	require( 'lib/wrap-es6-functions' )();
+}
+
+/**
+ * External dependencies
+ */
+import * as common from './common';
+import debugFactory from 'debug';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import createReduxStoreFromPersistedInitialState from 'state/initial-state';
+import detectHistoryNavigation from 'lib/detect-history-navigation';
+import userFactory from 'lib/user';
+
+const debug = debugFactory( 'calypso' );
+
+const withProject = () => {
+	// TODO: make project name dynamic based on config
+	const project = require( './project/wordpress-com' );
+
+	return ( funcName, ...params ) => {
+		common[ funcName ]( ...params );
```

Comment:
I agree it's less transparent, but maybe that can be mitigated by constructing `invoke` inline right before the invocations?

To me what's bothering isn't the fact that there are two statements per action, but that the statements have very different shapes — namely, the action (e.g. `locales`) shifts from being a variable holding a function to being a string (`'locales'`) with arguments added separately.